src/back/routes/badges.ts

```diff
@@ -72,3 +74,24 @@ async function revokeBadge(req: WithAuth): Promise<string> {
     return `Failed to revoke badges ${JSON.stringify(e)}`
   }
 }
+
+export type UploadResult = { status: ActionStatus; badgeCid: string; error?: string }
+async function uploadBadge(req: WithAuth): Promise<UploadResult> {
+  const user = req.auth!
```

Comment:
Is the `!` necessary? If you need to validate it we can add the validation in line 81.